prow/report/report.go

```diff
@@ -53,14 +53,17 @@ type configAgent interface {
 // Same goes for failed status.
 func reportStatus(ghc GithubClient, pj kube.ProwJob, cd string) error {
 	refs := pj.Spec.Refs
-	if err := ghc.CreateStatus(refs.Org, refs.Repo, refs.Pulls[0].SHA, github.Status{
-		State:       string(pj.Status.State),
-		Description: pj.Status.Description,
-		Context:     pj.Spec.Context,
-		TargetURL:   pj.Status.URL,
-	}); err != nil {
-		return err
+	if pj.Spec.Report {
+		if err := ghc.CreateStatus(refs.Org, refs.Repo, refs.Pulls[0].SHA, github.Status{
+			State:       string(pj.Status.State),
+			Description: pj.Status.Description,
+			Context:     pj.Spec.Context,
+			TargetURL:   pj.Status.URL,
+		}); err != nil {
+			return err
+		}
 	}
+
 	// Updating Children
 	if pj.Status.State != kube.SuccessState {
 		for _, nj := range pj.Spec.RunAfterSuccess {
```

Comment:
You probably want to gate these too.

Reply:
I thought about this, but is it not possible to have a parent that does not report, but have children that do? One use case would be you have a parent job that does some setup for its children, you might not want this to be reported, but you do want the children to report.

Reply:
Or did I completely misunderstand your ask ?

Reply:
All combinations are possible today ("parents report, children don't", "parents don't report, children report", "parents report, children report", "parents don't report, children don't report").

Reply:
That's my understanding as well. So what should I gate ?

Reply:
The following jobs about that we are about to send reports.
```go
	// Updating Children
 	if pj.Status.State != kube.SuccessState {
 		for _, nj := range pj.Spec.RunAfterSuccess {
 			cpj := pjutil.NewProwJob(nj)
            if !cpj.Spec.Report {
                continue
            }
 			cpj.Status.State = pj.Status.State
 			cpj.Status.Description = cd
 			cpj.Spec.Refs = refs
 			if err := reportStatus(ghc, cpj, cd); err != nil {
 				return err
 			}
 		}
```

Reply:
If you skip then you are not recursively going to the child. ie you would skip a child because a parent says to skip report.

Reply:
In my example, I am using the child's report field, not the parent's.

Reply:
Same issue, you are preventing the child's child job from having a report. The way I wrote it, works correclty as we talked. I don't understand the issue.

Reply:
Yeah, this is fine.